### PR TITLE
master-ponzi-getPeyout

### PR DESCRIPTION
**Pull request by arturtigranyan**

### CODE REVIEW

File: contracts/PonziGame.sol [hunk 1]
```diff
@@ -2,98 +2,31 @@ pragma solidity ^0.4.20;
...
-        investors[msg.sender] += msg.value;
-    }    
-
-    function getPaidByFactor(uint256 amount) public {
-           
-    }
-
-    function getPaidByOrder() public {
-        
+    struct Investments {
```
Comment (levon-hayrapetyan) [inline contracts/PonziGame.sol]: This structure represents a single Investment. Please consider removing the trailing "s"

File: contracts/PonziGame.sol [hunk 2]
```diff
@@ -2,98 +2,31 @@ pragma solidity ^0.4.20;
...
-    function readCurrentBalance() public returns (uint256) {
-        return this.balance;
-    }
-
-    function readTotalInvestment() public {
-
-    }
-
-    function readPayouts() public {
+    Investments[] investment;
```
Comment (levon-hayrapetyan) [inline contracts/PonziGame.sol]: the array represents multiple investments. Please consider naming it investment**s**.

File: contracts/PonziGame.sol [hunk 3]
```diff
@@ -2,98 +2,31 @@ pragma solidity ^0.4.20;
...
 
-    }
-    
-    function readNumberOfInvestors() public {
+    function getPaidByInvestmentOrder() payable {
```
Comment (levon-hayrapetyan) [inline contracts/PonziGame.sol]: As discussed yesterday I think "Invest" describes the behaviour of this function better than "getPaidByInvestmentOrder". Please consider renaming.

File: contracts/PonziGame.sol [hunk 4]
```diff
@@ -2,98 +2,31 @@ pragma solidity ^0.4.20;
...
-    function readCurrentPayout() public {
-
-    }
-
-    function readCurrentFee() public {
-
-    }
-
-    function bossTerminateContract() public {
+        investment.push(Investments(msg.sender, msg.value));
```
Comment (levon-hayrapetyan) [inline contracts/PonziGame.sol]: Please consider checking that the msg.value is not 0. We want to make sure that the investors invest something. In fact there is a requirement in the spec that investors should not be able to invest <100 wei.

File: contracts/PonziGame.sol [hunk 5]
```diff
@@ -2,98 +2,31 @@ pragma solidity ^0.4.20;
...
-    function selfDestroying() public {
+        boss.transfer(msg.value * 5 / 100);
         
+        while(investment[top].amount * 110 / 100  < this.balance){
+            investment[top].investor.transfer(investment[top].amount * 110 / 100);
+            top++;
+        }
+    }
+    
+     function getEthBalance(address _addr) constant returns(uint) {
```
Comment (levon-hayrapetyan) [inline contracts/PonziGame.sol]: why is this method needed? If to check the balance of the contract then you can simply return this.balance. I think the _addr parameter is redundant.

File: contracts/PonziGame.sol [hunk 6]
```diff
@@ -2,98 +2,33 @@ pragma solidity ^0.4.20;
...
+    
+    Investment[] investments;
 
-    function rankedOnTotalInvestment() public {        
+    function invest() payable {
         
-    }
-
-    function vipInvestor() public {
+        require(msg.value > 100);
```
Comment (martikyan) [inline contracts/PonziGame.sol]: Consider allowing investment to be 100 wei

**Pull request merged**